common/reactive/src/main/java/io/helidon/common/reactive/MultiFromOutputStream.java

```diff
@@ -116,48 +102,6 @@ public void write(int b) throws IOException {
     @Override
     public void close() throws IOException {
         complete();
-
-        if (!written.get() && !completionResult.isCompletedExceptionally()) {
-            // no need to wait for
-            return;
-        }
-        try {
-            completionResult.get(timeout, TimeUnit.MILLISECONDS);
-        } catch (CancellationException e) {
-            throw new IOException(e);
-        } catch (InterruptedException e) {
-            Thread.currentThread().interrupt();
-            throw new IOException(e);
-        } catch (ExecutionException e) {
-            throw new IOException(e.getCause());
-        } catch (TimeoutException e) {
-            throw new IOException("Timed out while waiting for subscriber to read data", e);
-        }
-    }
-
-    /**
-     * Signals this publishing output stream that it can safely return from otherwise blocking invocation
-     * to it's {@link #close()} method.
-     * Subsequent multiple invocations of this method are allowed, but have no effect on this publishing output stream.
-     * <p>
-     * When the {@link #close()} method on this output stream is invoked, it will block waiting for a signal to complete.
-     * This is useful in cases, when the receiving side needs to synchronize it's completion with the publisher, e.g. to
-     * ensure that any resources used by the subscribing party are not released prematurely due to a premature exit from
-     * publishing output stream {@code close()} method.
-     * <p>
-     * Additionally, this mechanism can be used to propagate any downstream completion exceptions back to this publisher
-     * and up it's call stack. When a non-null {@code throwable} parameter is passed into the method, it will be wrapped
-     * in an {@link IOException} and thrown from the {@code close()} method when it is invoked.
-     *
-     * @param throwable represents a completion error condition that should be thrown when a {@code close()} method is invoked
-     *                  on this publishing output stream. If set to {@code null}, the {@code close()} method will exit normally.
-     */
-    void signalCloseComplete(Throwable throwable) {
```

Comment:
@romain-grecourt Is this still actual? There is no exception downstream can propagate to upstream except NPO, also is blocking of close still neccessary? Only proper way to complete is from upstream's side or maybe wait for cancel from downstream, but in such case we cant complete from upstream with close which is blocked by waiting for cancel.

Reply:
Not sure why you are asking me, did I write this code ? :D

Reply:
Sorry I was following the blame :detective: . Next one to ask is @tomas-langer who modified the file last, WDYT Tomas?